resources/app/lr-aggregator.yaml

```diff
@@ -0,0 +1,124 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: lr-aggregator
+  namespace: kube-system
+  labels:
+    app: lr-aggregator
+spec:
+  type: ClusterIP
+  ports:
+    - name: rpc
+      protocol: TCP
+      port: 9966
+  selector:
+    app: lr-aggregator
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: lr-aggregator
+  namespace: kube-system
+  labels:
+    app: lr-aggregator
+data:
+  logrange.json: |
+    {
+      "HostHostId": 0,
+      "HostLeaseTTLSec": 5,
+      "HostRegisterTimeoutSec": 0,
+      "PublicApiRpc": {
+        "ListenAddr": "0.0.0.0:9966",
+        "TlsEnabled": false,
+        "Tls2Way": false,
+        "TlsCertFile": "/opt/logrange/config/tls/server.crt",
+        "TlsKeyFile": "/opt/logrange/config/tls/server.key",
+        "TlsCAFile": "/opt/logrange/config/tls/ca.pem"
+      },
+      "JrnlCtrlConfig": {
+        "JournalsDir": "/var/lib/logrange/db/",
+        "MaxOpenFileDescs": 500,
+        "CheckFullScan": false,
+        "RecoverDisabled": false,
+        "RecoverLostDataOk": false,
+        "WriteIdleSec": 120,
+        "WriteFlushMs": 500,
+        "MaxChunkSize": 5000000,
+        "MaxRecordSize": 1048576
+      },
+      "PipesConfig": {
+        "EnsureAtStart": [
+          {
+            "Name": "__default__"
+          }
+        ]
+      }
+    }
+
+  log4g.properties: |
+    # log4g configuration`
+
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: lr-aggregator
+  namespace: kube-system
+  labels:
+    app: lr-aggregator
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: lr-aggregator
+  template:
+    metadata:
+      labels:
+        app: lr-aggregator
+      annotations:
+        seccomp.security.alpha.kubernetes.io/pod: docker/default
+
+    spec:
+      securityContext:
+        runAsUser: 0
+      containers:
+        - name: lr-aggregator
+          image: "index.docker.io/logrange/logrange:v0.0.4"
+          imagePullPolicy: Always
+          ports:
+            - name: rpc
+              protocol: TCP
+              containerPort: 9966
+          volumeMounts:
+            - name: config
+              mountPath: /opt/logrange/config
+            - name: extdockercontainers
+              mountPath: /ext/docker/containers
+            - name: varlib
+              mountPath: /var/lib
+            - name: varlog
+              mountPath: /var/log
+      volumes:
+        - name: config
+          configMap:
+            name: lr-aggregator
+        - name: extdockercontainers
+          hostPath:
+            path: /ext/docker/containers
+        - name: varlib
+          hostPath:
+            path: /var/lib
```

Comment:
Do you need to mount the whole `/var/lib`? Is that where logrange database is stored, `/var/lib/logrange`? If so, I would only mount that.

Same for all other deployments.

Reply:
mmm, this is the question, speaking truly not sure... by default k8s node will not have this /var/lib/logrange folder and when Logrange starts it creates it in /var/lib, so it will be stored on Node (hopefully) and if I mount just /var/lib/logrange it will be just container folder as I understand (i.e. won't survive pod redeploy), might be I'm wrong or there is a way to specify otherwise during mount...

Reply:
ok, I think I got my answer, will switch to /var/lib/logrange then... "When you use a bind mount, a file or directory on the host machine is mounted into a container. The file or directory is referenced by its full path on the host machine. The file or directory does not need to exist on the Docker host already. It is created on demand if it does not yet exist. "